compose/foundation/foundation/src/uikitMain/kotlin/androidx/compose/foundation/gestures/UIKitTapGestureDetector.kt

```diff
@@ -0,0 +1,128 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.foundation.gestures
+
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.input.pointer.AwaitPointerEventScope
+import androidx.compose.ui.input.pointer.PointerEvent
+import androidx.compose.ui.input.pointer.PointerEventPass
+import androidx.compose.ui.input.pointer.PointerInputChange
+import androidx.compose.ui.input.pointer.PointerInputScope
+import androidx.compose.ui.input.pointer.PointerType
+import androidx.compose.ui.input.pointer.changedToDown
+import androidx.compose.ui.platform.ViewConfiguration
+import androidx.compose.ui.util.fastAll
+
+/**
+ * Detects repeating tap gestures on the given PointerInputScope.
+ *
+ * **Important**: for every tap sequence corresponding callback will be called.
+ *
+ * For example, if there is triple tap gesture, the call stack will look as follows: onTap, onDoubleTap, onTripleTap
+ *
+ * If there is a double tap gesture, the call stack will look as follows: onTap, onDoubleTap
+ *
+ * The order of callback execution is guaranteed.
+ *
+ * @param onTap  A callback function that will be called when a single tap gesture is detected. It will receive the position (Offset) where the tap occurred.
+ * @param onDoubleTap  A callback function that will be called when a double tap gesture is detected. It will receive the position (Offset) where the tap occurred.
+ * @param onTripleTap  A callback function that will be called when a triple tap gesture is detected. It will receive the position (Offset) where the tap occurred.
+ */
+internal suspend fun PointerInputScope.detectRepeatingTapGestures(
+    onTap: ((Offset) -> Unit)? = null,
+    onDoubleTap: ((Offset) -> Unit)? = null,
+    onTripleTap: ((Offset) -> Unit)? = null,
+) {
+    awaitEachGesture {
+        val touchesCounter = TouchCounter(viewConfiguration)
+        while (true) {
+            val down = awaitTouchEventDown()
+            touchesCounter.update(down)
+            val downChange = down.changes[0]
+            when (touchesCounter.touches) {
+                1 -> {
+                    if (onTap != null) {
+                        onTap(downChange.position)
+                        downChange.consume()
+                    }
+                }
+
+                2 -> {
+                    if (onDoubleTap != null) {
+                        onDoubleTap(downChange.position)
+                        downChange.consume()
+                    }
+                }
+
+                else -> {
+                    if (onTripleTap != null) {
+                        onTripleTap(downChange.position)
+                        downChange.consume()
+                    }
+                }
+            }
+        }
+    }
+}
+
+// Distance in pixels between consecutive click positions to be considered them as clicks sequence
+private const val TouchesSlop = 100.0
```

Comment:
It seems this was a mistake in the original mouse handling block. It works in pixels instead of `dp`. I would add it to `ClickCounter`, and use a different value, calculated this way:
```
ClickCounter(
    viewConfiguration = viewConfiguration,
    slop = 50.dp.toPx()
)
```

50 - probably should be different for iOS. Can we test native applications, and set a similar value?

Reply:
Seems like for iOS `50.dp` also works fine now. But I'll create a new issue to research this.